src/CommonCartridge.js

```diff
@@ -449,382 +469,398 @@ export default class CommonCartridge extends Component {
+                      </NavLink>
+                    )}
+                    {numberOfAssignments > 0 && (
+                      <NavLink
+                        className="MenuItem"
+                        to={{
+                          pathname: "/assignments",
+                          search: startIndexQuery
+                        }}
+                      >
+                        <Trans>Assignments ({numberOfAssignments})</Trans>
+                      </NavLink>
+                    )}
+                    {this.state.pageResources.length > 0 && (
+                      <NavLink
+                        className="MenuItem"
+                        to={{
+                          pathname: "/pages",
+                          search: startIndexQuery
+                        }}
+                      >
+                        <Trans>Pages ({this.state.pageResources.length})</Trans>
+                      </NavLink>
+                    )}
+                    {this.state.discussionResources.length > 0 && (
+                      <NavLink
+                        className="MenuItem"
+                        to={{
+                          pathname: "/discussions",
+                          search: startIndexQuery
+                        }}
+                      >
+                        <Trans
+                          id="Discussions ({length})"
+                          values={{
+                            length: this.state.discussionResources.length
                           }}
-                        >
-                          <Trans>
-                            Pages ({this.state.pageResources.length})
-                          </Trans>
-                        </NavLink>
+                        />
+                      </NavLink>
+                    )}
+                    {this.state.assessmentResources.length > 0 && (
+                      <NavLink
+                        className="MenuItem"
+                        to={{
+                          pathname: "/quizzes",
+                          search: startIndexQuery
+                        }}
+                      >
+                        <Trans>
+                          Quizzes ({this.state.assessmentResources.length})
+                        </Trans>
+                      </NavLink>
+                    )}
+                    {this.state.fileResources.length > 0 && (
+                      <NavLink
+                        className="MenuItem"
+                        to={{
+                          pathname: "/files",
+                          search: startIndexQuery
+                        }}
+                      >
+                        <Trans>Files ({this.state.fileResources.length})</Trans>
+                      </NavLink>
+                    )}
+                  </div>
+                  <Button
+                    as="label"
+                    variant="ghost"
+                    buttonRef={ref =>
```

Comment:
Same as above comment. Maybe I am missing something. I don't exactly know what buttonRef does but I suspect it's just a simple ref forwarding prop of the Button component. In that case I think this logic does not belong here.

Reply:
Same here, in order to have attributes and classes the ref is used (its a hack)